client/signup/navigation-link/test/index.jsx

```diff
@@ -45,42 +50,43 @@ describe( 'NavigationLink', () => {
 	} );
 
 	test( 'should render Button element', () => {
-		const wrapper = shallow( <NavigationLink { ...props } /> );
+		const { container } = render( <NavigationLink { ...props } /> );
 
-		expect( wrapper.find( 'ForwardRef(Button)' ) ).toHaveLength( 1 );
-		expect( wrapper.find( 'ForwardRef(Button)' ).props().className ).toEqual( 'navigation-link' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 1 );
+		expect( container.firstChild.nodeName ).toBe( 'BUTTON' );
+		expect( container.firstChild ).toHaveClass( 'navigation-link' );
```

Comment:
Let's use the above `screen` query for finding the button as well. We could abstract it to a constant.